bench/irmin-pack/tree.ml

```diff
@@ -0,0 +1,427 @@
+      inode_config = `Entries_32;
+      run = Bench_inodes_32.run_read_trace;
+    };
+    {
+      mode = `Chains;
+      speed = `Quick;
+      inode_config = `Entries_32;
+      run = Bench_inodes_32.run ~mode:`Chains;
+    };
+    {
+      mode = `Chains;
+      speed = `Slow;
+      inode_config = `Entries_2;
+      run = Bench_inodes_2.run ~mode:`Chains;
+    };
+    {
+      mode = `Large;
+      speed = `Quick;
+      inode_config = `Entries_32;
+      run = Bench_inodes_32.run ~mode:`Large;
+    };
+    {
+      mode = `Large;
+      speed = `Slow;
+      inode_config = `Entries_2;
+      run = Bench_inodes_2.run ~mode:`Large;
+    };
+  ]
+
+let pp_inode_config fmt = function
+  | `Entries_2 -> Format.fprintf fmt "[2, 5]"
+  | `Entries_32 -> Format.fprintf fmt "[32, 256]"
+
+let pp_config b config fmt () =
+  match b.mode with
+  | `Read_trace ->
+      let stats = Bench_inodes_32.Trees_trace.get_stats () in
+      Format.fprintf fmt
+        "Tezos_log mode on inode config %a: %d commits (%d adds; %d removes; \
+         %d finds; %d mems; %d mem_tree; %d copies)"
+        pp_inode_config b.inode_config config.ncommits_trace stats.adds
+        stats.removes stats.finds stats.mems stats.mem_tree stats.copies
+  | `Chains ->
+      Format.fprintf fmt
+        "Chain trees mode on inode config %a: %d commits, each consisting of \
+         %d chains of depth %d"
+        pp_inode_config b.inode_config config.ncommits config.nchain_trees
+        config.depth
+  | `Large ->
+      Format.fprintf fmt
+        "Large trees mode on inode config %a: %d commits, each consisting of \
+         %d large trees of %d entries"
+        pp_inode_config b.inode_config config.ncommits config.nlarge_trees
+        config.width
+
+let main ncommits ncommits_trace operations_file quick depth width nchain_trees
+    nlarge_trees =
+  let config =
+    {
+      ncommits;
+      ncommits_trace;
+      operations_file;
+      root = "test-bench";
+      quick;
+      depth;
+      width;
+      nchain_trees;
+      nlarge_trees;
+    }
+  in
+  Printexc.record_backtrace true;
+  Random.self_init ();
+  FSHelper.rm_dir config.root;
+  let suite =
+    if config.quick then List.filter (fun b -> b.speed = `Quick) suite
+    else (* only run first element of the suite when quick *) List.tl suite
+  in
+  let run_benchmarks () =
+    Lwt_list.fold_left_s
+      (fun (config, results) (b : suite_elt) ->
+        let+ config, result = b.run config in
+        (config, (b, result) :: results))
+      (config, []) suite
+  in
+  let config, results = Lwt_main.run (run_benchmarks ()) in
+  let pp_result fmt (b, result) =
+    Format.fprintf fmt "Configuration:@\n @[%a@]@\n@\nResults:@\n @[%a@]@\n"
+      (pp_config b config) () Benchmark.pp_results result
+  in
+  Fmt.pr "%a@." Fmt.(list ~sep:(any "@\n@\n") pp_result) results
+
+open Cmdliner
+
+let quick =
+  let doc = Arg.info ~doc:"Run the quick benchmarks" [ "quick" ] in
+  Arg.(value @@ flag doc)
```

Comment:
I'll need a finer control over what's executed for the profiling tests that I'm currently running, but I'll PR on top of you.